website/members/views.py

```diff
@@ -321,6 +321,162 @@ def get(self, request):
+                return HttpResponse("Error al armar la página")
+
+            # if missing the signed letter, produce it and attach it
+            if missing_info['missing_signed_letter']:
+                letter_filepath = self._generate_letter(member)
+
+            # build the mail
+            recipient = f"{member.entity.full_name} <{member.entity.email}>"
+            mail = EmailMessage(
+                self.MAIL_SUBJECT, text, self.MAIL_FROM, [recipient],
+                cc=[self.MAIL_MANAGER], reply_to=[self.MAIL_MANAGER])
+            if missing_info['missing_signed_letter']:
+                mail.attach_file(letter_filepath)
+
+            # actually send it
+            try:
+                mail.send()
+            except Exception as err:
+                sent_error += 1
+                logger.error(
+                    "Problems sending email [%s] to member %s: %r", errors_code, member, err)
+            else:
+                sent_ok += 1
+            finally:
+                if missing_info['missing_signed_letter']:
+                    try:
+                        os.unlink(letter_filepath)
+                    except Exception as exc:
+                        logger.warning("Couldn't remove letter file %r: %r", letter_filepath, exc)
+
+        deltat = time.time() - tini
+        context = {
+            'sent_ok': sent_ok,
+            'sent_error': sent_error,
+            'errors_code': errors_code,
+            'deltamsec': int(deltat * 1000),
+        }
+        return render(request, 'members/mail_sent.html', context)
+
+    def _analyze_member(self, member):
+        """Analyze and indicate in which categories the member is missing somethhing."""
+        cat_student = Category.objects.get(name=Category.STUDENT)
+        cat_collab = Category.objects.get(name=Category.COLLABORATOR)
+        # simple flags with "Not Applicable" situation
+        missing_student_certif = (
+            member.category == cat_student and member.has_student_certificate)
+
+        missing_collab_accept = (
+            member.category == cat_collab and member.has_collaborator_acceptance)
+
+        # info from Person
+        missing_nickname = member.person.nickname != ""
+        # picture is complicated, bool() is used to check if the Image field has an associated
+        # filename, and False itself is used as the "dont want a picture!" flag
+        picture = member.person.picture is not None
+
+        # info from Member itself
+        missing_payment = member.first_payment_month is not None and member.category.fee != 0
+        missing_signed_letter = member.has_subscription_letter
+        missing = missing_signed_letter and missing_nickname and picture
+
+        return {
+            'missing': missing,
+            'missing_student_certif': missing_student_certif,
+            'missing_payment': missing_payment,
+            'missing_collab_accept': missing_collab_accept,
+        }
```

Comment:
No point in returning a complex structure here, all it needs to know if the member has everything complete and ready to be approved, or not. A bool is enough.